presto-parquet/src/main/java/io/prestosql/parquet/reader/MetadataReader.java

```diff
@@ -314,55 +313,58 @@ private static PrimitiveTypeName getTypeName(Type type)
-                return OriginalType.ENUM;
+                return LogicalTypeAnnotation.enumType();
             case DECIMAL:
-                return OriginalType.DECIMAL;
+                int scale = (schemaElement == null ? 0 : schemaElement.scale);
+                int precision = (schemaElement == null ? 0 : schemaElement.precision);
+                return LogicalTypeAnnotation.decimalType(scale, precision);
             case DATE:
-                return OriginalType.DATE;
+                return LogicalTypeAnnotation.dateType();
             case TIME_MILLIS:
-                return OriginalType.TIME_MILLIS;
+                return LogicalTypeAnnotation.timeType(true, LogicalTypeAnnotation.TimeUnit.MILLIS);
+            case TIME_MICROS:
+                return LogicalTypeAnnotation.timeType(true, LogicalTypeAnnotation.TimeUnit.MICROS);
             case TIMESTAMP_MILLIS:
-                return OriginalType.TIMESTAMP_MILLIS;
+                return LogicalTypeAnnotation.timestampType(true, LogicalTypeAnnotation.TimeUnit.MILLIS);
+            case TIMESTAMP_MICROS:
+                return LogicalTypeAnnotation.timestampType(true, LogicalTypeAnnotation.TimeUnit.MICROS);
             case INTERVAL:
-                return OriginalType.INTERVAL;
+                return LogicalTypeAnnotation.IntervalLogicalTypeAnnotation.getInstance();
             case INT_8:
-                return OriginalType.INT_8;
+                return LogicalTypeAnnotation.intType(8, true);
             case INT_16:
-                return OriginalType.INT_16;
+                return LogicalTypeAnnotation.intType(16, true);
             case INT_32:
-                return OriginalType.INT_32;
+                return LogicalTypeAnnotation.intType(32, true);
             case INT_64:
-                return OriginalType.INT_64;
+                return LogicalTypeAnnotation.intType(64, true);
             case UINT_8:
-                return OriginalType.UINT_8;
+                return LogicalTypeAnnotation.intType(8, false);
             case UINT_16:
-                return OriginalType.UINT_16;
+                return LogicalTypeAnnotation.intType(16, false);
             case UINT_32:
-                return OriginalType.UINT_32;
+                return LogicalTypeAnnotation.intType(32, false);
             case UINT_64:
-                return OriginalType.UINT_64;
+                return LogicalTypeAnnotation.intType(64, false);
             case JSON:
-                return OriginalType.JSON;
+                return LogicalTypeAnnotation.jsonType();
             case BSON:
-                return OriginalType.BSON;
-            case TIMESTAMP_MICROS:
-                return OriginalType.TIMESTAMP_MICROS;
-            case TIME_MICROS:
-                return OriginalType.TIME_MICROS;
+                return LogicalTypeAnnotation.bsonType();
             default:
-                throw new IllegalArgumentException("Unknown converted type " + type);
```

Comment:
The original error message seems good. IAE is correct since this is a bad argument.

Reply:
This is verbatim copy of `org.apache.parquet.format.converter.ParquetMetadataConverter#getLogicalTypeAnnotation`, I'd prefer to keep it this way.

This should never be raised anyway, so likely it doesn't matter.